cmd/srcd/cmd/utils_test.go

```diff
@@ -0,0 +1,216 @@
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type logMessage struct {
+	Msg   string
+	Time  string
+	Level string
+}
+
+type DeferedTestSuite struct {
+	suite.Suite
+}
+
+func TestDeferedTestSuite(t *testing.T) {
+	suite.Run(t, new(DeferedTestSuite))
+}
+
+var newLineRegex = regexp.MustCompile(`\r?\n`)
+
+func (s *DeferedTestSuite) traceLogMessages(fn func(), memLog *bytes.Buffer) []logMessage {
+	s.T().Helper()
+
+	logrus.SetOutput(memLog)
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+
+	fn()
+
+	var dat []logMessage
+	if memLog.Len() == 0 {
+		return dat
+	}
+
+	bytes := memLog.Bytes()
+	bytes = newLineRegex.ReplaceAll(bytes, []byte(","))
+	bytes = bytes[:len(bytes)-1]
+
+	start, end := "[", "]"
+	bytes = append(bytes, end...)
+	bytes = append([]byte(start), bytes...)
+
+	if err := json.Unmarshal(bytes, &dat); err != nil {
```

Comment:
that's an interesting solution. But usually, with this problem people deal like this: https://play.golang.org/p/gaeRD1k7aPw

Reply:
That's much better! Thanks for the example!